### PR TITLE
link-redirect-fix

### LINKED ISSUE #1038: 🐛 Bug: Mis-linked page (Again)
### Describe the bug

This line of code 
```
const markdownFile = '_indexPage';
```
is faulty, every page with this code which has a "Contribute to this on GitHub" redirects to https://github.com/json-schema-org/website/blob/main/pages/{url}/_index.md , which is a 404.

### Steps To Reproduce

Example:
1. Navigate to https://json-schema.org/overview/faq
2. Click on the Contribute to GitHub button
3. Arrive at https://github.com/json-schema-org/website/blob/main/pages/overview/faq/_index.md (404)
4. Error reproduced

### Expected Behavior

A 200 response on the editable source code page.

### Screenshots

_No response_

### Device Information [optional]

```markdown
- OS: Ubuntu 20.04 Noble
- Browser: Firefox, Chrome
- version: 22
```

### Are you working on this issue?

Yes

### Do you think this work might require an [Architectural Decision Record (ADR)]? (significant or noteworthy)

No

### PR DESCRIPTION
**Pull request by Yashwanth1906**
What kind of change does this PR introduce?

Bugfix: Fixes misdirection of pages.
Issue Number:

Closes #1038 
Related to #___ 
Others? 
Screenshots/videos:

If relevant, you can add screenshots or videos showing the issue and how it was fixed.
If relevant, did you update the documentation?

No documentation updates were needed.
Summary

 
This PR fixes a bug that caused misdirection of certain pages in the app. The routing logic was adjusted to correctly map the pages to their intended destinations.
Does this PR introduce a breaking change?

No, this PR does not introduce a breaking change.

### CODE REVIEW

File: package-lock.json
Comment (DhairyaMajmudar) [inline package-lock.json:1]: Pls. remove `package-lock.json` file since we are using yarn as package manager

File: package.json
Comment (DhairyaMajmudar) [inline package.json:1]: remove newly installed dependencies since it seems like they are not used in the codebase

**Comment by IyaadDev** [comment]
looks great, but would it be possible to clarify:
-  the purposes of 'pages/overview/code-of-conduct/_index.md' and 'pages/overview/sponsors/_index.md',
- the modifications to project dependencies
- the new hyper-schema section

File: pages/overview/code-of-conduct/_index.md
Comment (DhairyaMajmudar) [inline pages/overview/code-of-conduct/_index.md:1]: This file is obtained from [.github](https://github.com/json-schema-org/.github/blob/main/CODE_OF_CONDUCT.md) repo as submodule, for doing changes in it you have to perform changes in that repo.  

Revert the changes related to it.

File: pages/overview/sponsors/_index.md
Comment (DhairyaMajmudar) [inline pages/overview/sponsors/_index.md:1]: This file is obtained from [community](https://github.com/json-schema-org/community) repo as git submodule 
Pls. revert changes related to it.

File: public/draft-00/hyper-schema
Comment (DhairyaMajmudar) [inline public/draft-00/hyper-schema:1]: Pls. revert these changes

File: yarn.lock
Comment (DhairyaMajmudar) [inline yarn.lock:1]: remove changes from yarn.lock file and uninstall newly installed dependencies

**Review by DhairyaMajmudar — CHANGES_REQUESTED** [review]

**Comment by Yashwanth1906** [comment]
> looks great, but would it be possible to clarify:
> 
> * the purposes of 'pages/overview/code-of-conduct/_index.md' and 'pages/overview/sponsors/_index.md',
> * the modifications to project dependencies
> * the new hyper-schema section

**Comment by Yashwanth1906** [comment]
Ill Fork the repo again and create a new pull request soon @DhairyaMajmudar . Thank you

**Comment by DhairyaMajmudar** [comment]
> Ill Fork the repo again and create a new pull request soon @DhairyaMajmudar . Thank you

No need to fork and create pr again you can commit changes in this PR also
